File: app/src/main/java/com/adapter/MyViewPagerAdapter.java
package com.adapter;

import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.Drawable;
import android.media.Image;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.viewpager.widget.PagerAdapter;

import com.bumptech.glide.Glide;
import com.model.ProductResponse;
import com.solidindia.R;

import java.util.List;

public class MyViewPagerAdapter extends PagerAdapter {
    private LayoutInflater layoutInflater;
    private Context context;
    List<ProductResponse.Data.Category.Product.ProductImage> introSliderList;
    public MyViewPagerAdapter(Context context,
                              List<ProductResponse.Data.Category.Product.ProductImage> introSliderList) {
        layoutInflater = LayoutInflater.from(context);
        this.introSliderList = introSliderList;
        this.context = context;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        View view = layoutInflater.inflate(R.layout.row_slider, container, false);
        ProductResponse.Data.Category.Product.ProductImage item = introSliderList.get(position);
        ImageView ivSilder = (ImageView) view.findViewById(R.id.ivSilder);

        Glide.with(context)
                    .load(item.getPath())
                    .error(R.mipmap.ic_launcher)
                    .into(ivSilder);


        //view.setBackgroundColor(Color.parseColor(item.getBackgroundColor())/*colors[position]*/);
        container.addView(view);
        return view;
    }

    @Override
    public int getCount() {
        return introSliderList.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object obj) {
        return view == obj;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        View view = (View) object;
        container.removeView(view);
    }
}

File: app/src/main/java/com/adapter/MyViewPagerAdapter.kt
package com.adapter

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.viewpager.widget.PagerAdapter
import com.app.AppController
import com.bumptech.glide.Glide
import com.github.tntkhang.fullscreenimageview.library.TouchImageView
import com.model.ProductResponse
import com.solid1972.R
import com.solid1972.activity.ShowFullImageActivity
import com.utils.loadImage


class MyViewPagerAdapter(
    private val activity: Activity,
    private val context: Context,
    internal var introSliderList: ArrayList<ProductResponse.Data.Category.Product.ProductImage>,
    var isClickViewPagerImage: Boolean
) : PagerAdapter() {
    private val layoutInflater: LayoutInflater
    lateinit var appcontroller: AppController

    init {
        appcontroller = activity.application as AppController
        layoutInflater = LayoutInflater.from(context)
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view: View
        if (isClickViewPagerImage) {
            view = layoutInflater.inflate(R.layout.row_slider, container, false)
            val (path, type) = introSliderList[position]
            val ivSilder = view.findViewById<View>(R.id.ivSilder) as ImageView

            loadImage(path, appcontroller.getAppContext()!!, ivSilder, R.mipmap.ic_launcher)

            if (isClickViewPagerImage) {
                ivSilder.setOnClickListener(object : View.OnClickListener {
                    override fun onClick(p0: View?) {
                        val intent = Intent(context, ShowFullImageActivity::class.java)
                        intent.putExtra("position", position)
                        intent.putParcelableArrayListExtra("sliderList", introSliderList)
                        context.startActivity(intent)
                    }
                })
            }
        } else {
            view = layoutInflater.inflate(R.layout.row_slider_full_image, container, false)
            val ivFullImage = view.findViewById<View>(R.id.iv_content) as TouchImageView
            val (path, type) = introSliderList[position]

            //        TouchImageView ivContent = view.findViewById(R.id.iv_content);

            ivFullImage.setZoom(0.99f);
            Glide.with(appcontroller.getAppContext()!!)
                    .load(path)
                    .into(ivFullImage)

            //  loadImage(path, appcontroller.getAppContext()!!, ivFullImage, R.mipmap.ic_launcher)
        }

        /* Glide.with(context)
                    .load(item.getPath())
                    .error(R.mipmap.ic_launcher)
                    .into(ivSilder);*/


        //view.setBackgroundColor(Color.parseColor(item.getBackgroundColor())/*colors[position]*/);
        container.addView(view)
        return view
    }

    override fun getCount(): Int {
        return introSliderList.size
    }

    override fun isViewFromObject(view: View, obj: Any): Boolean {
        return view === obj
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        val view = `object` as View
        container.removeView(view)
    }
}
